component/prometheus/exporter/agent/agent.go

```diff
@@ -0,0 +1,42 @@
+package agent
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus/exporter"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/agent"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "prometheus.exporter.agent",
+		Args:          Arguments{},
+		Exports:       exporter.Exports{},
+		Singleton:     true,
```

Comment:
We should remove this singleton flag. It is going to be going away soon, and there isn't a hard technical reason you can't have 2 of these (even if it is silly :))

Reply:
Singleton removed. Thanks for the context! 😄